### ISSUE TITLE
Provide link in Analytics settings panel to update GA4 configuration and improve UI for related UA link

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3702]
In the Analytics settings panel (in view-only state), there is currently a long message with a link that points to the Analytics frontend for the currently active Analytics property and its view.

Now that users can also have a GA4 property configured there, this message is confusing, as it appears below all of those settings while still only focusing on UA. We will now need two links instead of just the one. In addition, it's a bit strange to have a long sentence as a complete link anyway. Let's revisit this UI.

Screenshot:
<img width="647" alt="Screen Shot 2021-07-13 at 9 55 56 AM" src="https://user-images.githubusercontent.com/3531426/125494185-71b2e7f5-5c2d-47ea-9a3e-9563ec5f6782.png">

A potential solution would be:
* Display a small "Edit (Universal Analytics configuration) in Analytics" external link for the UA property on the right of the three UA values.
* Display a similar small "Edit (Google Analytics 4 configuration) in Analytics" external link for the GA4 property on the right of the two UA values.

cc @Pratheep-lab 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

In the Analytics settings view:
* Remove the link "You can make changes to this view..." and instead place a short link (with external indicator) to the same URL on the right side of the "View" value.
    * It should link to `https://analytics.google.com/analytics/web/?authuser={googleEmailAddress}#/a{accountID}w{uaPropertyID}p{viewID}/admin/view/settings` (same as the previous "You can make changes to this view..." link)
    * It should say "Edit in Analytics"
    * For screen readers, it should be more specific: "Edit Universal Analytics configuration in Analytics"
* If a GA4 property is also connected, display a similar link on the right side of the "Measurement ID" value:
    * It should link to `https://analytics.google.com/analytics/web/?authuser={googleEmailAddress}#/a{accountID}p{ga4PropertyID}/admin/streams/table/{webDataStreamID}`
    * It should say "Edit in Analytics"
    * For screen readers, it should be more specific: "Edit Google Analytics 4 configuration in Analytics"
* The links should both appear in the same vertical position and with the same font-size as the individual setting values (i.e. the IDs that display right under all the labels).
* The changes should be available/previewable in Storybook.

## Implementation Brief

* Continue and merge PR here https://github.com/google/site-kit-wp/pull/3766

Using `assets/js/modules/analytics/components/settings/SettingsView.js`

In order to build the GA4 Url
* Get the `webDataStreamID` from `select( MODULES_ANALYTICS_4 ).getWebDataStreamID()`
 * Build the Url like `editViewSettingsURL` but with the same Url structure as the AC (passing this into `getServiceURL`)

* Update the text for the existing UA link as per the AC

These links should be encoded with `escapeURI` (requires updating existing UA link)

* Move this link up into the same div `.googlesitekit-settings-module__meta-items` as containing the UA info
* Create a new link similar to this and add it to the end of the same div as the `.googlesitekit-settings-module__meta-items` of the GA4 info
* Use the new GA4 link here
* Set the text for this GA4 link as per the AC

In order to line up the text for both these links
* add empty `<h5>`s at the top of each Link's `googlesitekit-settings-module__meta-item` (just containing a `&nbsp;`).
* these `<h5>`s should be hidden on mobile

To style the Links to be the same size as the rest of the text
* add a new class `.googlesitekit-settings-module__inline-link` to the Links (through the prop className)
* Add a new rule to `assets/sass/components/settings/_googlesitekit-settings-module.scss`
```
.googlesitekit-settings-module__inline-link {
    font-size: 1rem;
}
```

 

### Test Coverage

* There should be no changes to existing tests
* 
### Visual Regression Changes

* There should be no changes to existing VRTs

## QA Brief

* Connect Site Kit to an Analytics account that has a GA4 and a UA property
* Go to Analytics Settings page
* Observe new changes with new links as per AC

## Changelog entry

* Add and update deep links on Analytics settings view to edit the UA property view and GA4 measurement ID.

**Comment by eclarke1** [comment]
@Pratheep-lab I've created a task in Asana for this https://app.asana.com/0/1185360131189553/1200629922083182 where we can handle the UX recommendations.

**Comment by felixarntz** [comment]
@marrrmarrr I defined ACs based on what we had discussed 1-2 weeks ago, let's for now display two short "Edit in Analytics" links right next to the Analytics IDs.

cc @Pratheep-lab @eclarke1 This should get a UX review during the implementation, e.g. using Storybook.

**Comment by danielgent** [comment]
![storybook-ok](https://user-images.githubusercontent.com/8496063/127150760-920eaf5d-5329-44fb-ae31-b0b3fcfe5d07.png)

The current storybook story is enough. I could add another one for when there *is no* GA4 property but that seems out-of-scope for this ticket

**Comment by eugene-manuilov** [comment]
@danielgent, please, avoid using "PR as IB" approach. I know it's a big temptation to start coding instead of writing IB but we need to write IB first because sometimes things that seem obvious may change during IB reviews.

We can use "PR as IB" approach only for edge cases when we found something small that needs to be squeezed into the current release. But for regular tickets, we should go with our main procedure and create comprehensive IBs.

Please, update this ticket to have standard IB. You can leave a link to your PR at the end of IB.

**Comment by danielgent** [comment]
The AC is already a detailed list of what steps to do

I'm just going to paste the code into the IB then :shrug: 

I think I understand the concept of an IB for complex tickets (although I can't find other people doing this online) but this diff is tiny

**Comment by danielgent** [comment]
I've just added an IB.  Either I've done it totally wrong, or I'm missing something, because it's *way* easier to read my annotated PR and look at the code, than read those bullet points :smile: 

It reminds me of podcasts trying to explain code and them lamenting that they could show it on a video much easier

**Comment by eugene-manuilov** [comment]
> * Build the Url for this as per the AC
> * Move this link up into the same div ...
> * Create a new link similar to this and add it to the end of the same div ...

@danielgent could you, please, be more specific? What do you mean by "Build the URL for this"? Which link do you mean when you say "Move this link up into the same div..."? IB needs to be clear, so everyone can understand what we need to do there.

> Build the Url for this as per the AC

When we compose URLs using string templates, we need to prepend the template with the `escapeURI` tag function to correctly encode parts of the URL. I know that there are some URL templates that don't use it, we will need to be fixed at some point. For now, please, update IB to mention that we need to use the `escapeURI` tag function for the compound path for the GA4 URL and to add it to the compound path for the existing UA URL.

> To style the Links to be the same size as the rest of the text

We need to wrap the link with `<p className="googlesitekit-settings-module__meta-item-data">` to make it have the same size.

> In order to line up the text for both these links
> * add empty <h5>s at the top of each Link's googlesitekit-settings-module__meta-item (just containing a &nbsp;)

We need to hide these <h5> tags on small screens to avoid displaying big gaps between the setting and the link.

**Comment by eugene-manuilov** [comment]
> For screen readers, it should be more specific: "Edit Google Analytics 4 web data stream in Analytics"

@felixarntz, as far as I remember, we didn't want users to face the "_web data stream_" terminology in our UI. Did we change our minds?

**Comment by eugene-manuilov** [comment]
> The AC is already a detailed list of what steps to do

Yes, sometimes AC contains too many technical details. But it doesn't mean that there is nothing to add to it in IB. We can refer to specific sections of AC when we write IB, but AC shouldn't be a replacement for our IB.

**Comment by felixarntz** [comment]
@eugene-manuilov 
> as far as I remember, we didn't want users to face the "web data stream" terminology in our UI. Did we change our minds?

That's a fair point. We may change this in the future, certainly if we're going to allow picking a web data stream (see #3615), but for now let's keep it out of here. I'm going to adjust the wording to be a more abstract "Edit ... configuration in Analytics".

**Comment by danielgent** [comment]
@eugene-manuilov  I've had another go at updating the IB

`Move this link up into the same div .googlesitekit-settings-module__meta-items as containing the UA info`

Honestly, I'm not sure what more I can write. I would just show the code change...

From the AC I would just do the PR (as I have done) and then we would discuss what I've done in the code review.

I've spent another hour trying to describe what I've done and I don't see it as any clearer than just reading the PR! :smile:  (as long as I spent writing the PR)

**Comment by eugene-manuilov** [comment]
Thanks, @danielgent. 

> From the AC I would just do the PR (as I have done) and then we would discuss what I've done in the code review.

This is not how it works on this project. When we write IBs, we can discuss the implementation, all possible pitfalls, and ask all questions before an engineer starts working on it. So when it comes to implementation, the engineer will know exactly what needs to be done and the scope of work that is required.

Your approach is easier for you, but it leaves all questions and pitfalls to the end where a code reviewer will need to figure out what needs to be done, what is missing, and so on. Plus, let's say you spent X hours on your PR, but what will happen if it turns out that what you did is not what we need or it is done in the wrong way? A part of those X hours (let's say 30%) will be spent on work that is useless and we will need to spend additional Y hours to rework it. I am not talking about this ticket and PR in particular, just trying to explain to you why IB matters.

> `Move this link up into the same div .googlesitekit-settings-module__meta-items as containing the UA info`
> Honestly, I'm not sure what more I can write. I would just show the code change...

My main concern is about phrases like `this link`, `use ... link here`, etc. It makes me guess which link you mean. If we can remove this uncertainty, it will be much easier to follow your IB. If we write like the following, it will be easier to understand, won't it?

```
Move UA link up into the same div .googlesitekit-settings-module__meta-items as containing the UA info
```
```
Use the new GA4 URL for the new GA4 link added to the end of the same div as the .googlesitekit-settings-module__meta-items of the GA4 info
```

> > To style the Links to be the same size as the rest of the text
> 
> We need to wrap the link with `<p className="googlesitekit-settings-module__meta-item-data">` to make it have the same size.

Looks like you missed this ^ from my comment: https://github.com/google/site-kit-wp/issues/3702#issuecomment-888269858. We need to wrap the link with the `<p>` tag which already has styles with the correct font size. The only thing that we need to add is to reset the `font-weight` property to have regular weight. No need to set the `font-size` property for the link, it will be inherited from the `<p>` tag.

**Comment by eclarke1** [comment]
Let's pick this one up offline team so we can align on an approach together cc @aaemnnosttv

**Comment by aaemnnosttv** [comment]
Thanks @danielgent and @eugene-manuilov – I think the IB pretty close and there are just a few minor details left to clarify here (mostly I think the added inline link class probably isn't necessary and we may want to move the exact position of the link in the markup). Seeing as we already have a PR in progress that is also quite close to what we need I'll move this forward and we can continue the conversation in code review 👍 

IB ✅

**Comment by wpdarren** [comment]
### QA Update: Pass ✅ 

**Verified:**

In Google Analytics settings (not edit mode): 

- The link `You can make changes to this view...` is removed
- In its place is a short link (with external indicator) to the right side of the `View` value.
- Confirmed that the URL is the same as before and format detailed in AC. 
- Confirmed the short link says `Edit in Analytics`
- Confirmed for screen readers, the title is more specific: `Edit Universal Analytics property view..`

With aGA4 property connected:

- The same link appears on the right side of the `Measurement ID` value 
- Confirmed that it links to the format URL within the AC. 
- Confirmed that It says  `Edit in Analytics` 
- Confirmed for screen readers, it should be more specific: `Edit Google Analytics 4 web data stream` 
- Confirmed that the links appear in the same vertical position and with the same font-size as the individual setting values.

Additional checks:

- Tested the UI changes on our supported browsers
- Tested the UI changes on Android and iOS devices 
- Checked that the UI changes are available in Storybook.

![image](https://user-images.githubusercontent.com/73545194/128191056-b7182fde-dcc1-451e-90e3-c32a30a1f9cf.png)

**Comment by felixarntz** [comment]
## Approval ❌ 

While this is not about something missed from the ACs or broken behavior, the UI implementation looks not satisfying at this point, since the text is literally right next to the field on its left (see screenshot below).

<img width="642" alt="Screen Shot 2021-08-12 at 9 14 11 AM" src="https://user-images.githubusercontent.com/3531426/129234061-672f95f3-6d1f-459d-8a0c-c9c285513edc.png">

I'm not sure why it was decided to put the new "Edit in Analytics" links into the same `.googlesitekit-settings-module__meta-item` container that is used for the last existing item - the edit links apply to the entire Analytics entity, not just the "UA view" or "GA4 web data stream", which they are now nested within.

The two edit links should instead be rendered within their own `.googlesitekit-settings-module__meta-item` containers, which will also automatically resolve the UI problem, adding the expected margin on the left.

**Comment by eugene-manuilov** [comment]
@felixarntz here is a follow-up PR: #3864. 

https://google.github.io/site-kit-wp/storybook/pull/3864/?path=/story/analytics-module-settings--view-open-with-all-settings-ga4

**Comment by felixarntz** [comment]
Looks a lot better now!
<img width="619" alt="Screen Shot 2021-08-12 at 3 46 07 PM" src="https://user-images.githubusercontent.com/3531426/129279771-c8b11150-9577-439f-8608-0c5df90c9f73.png">

**Comment by danielgent** [comment]
![unalligned](https://user-images.githubusercontent.com/8496063/129341650-1f0442d0-e45d-4eaa-9006-65ac2a598f26.png)

It's not aligned though @felixarntz  @eugene-manuilov @aaemnnosttv   :smile: 

On my original approach in the IB I used empty headers as a hack so that it all lined up

**Comment by wpdarren** [comment]
@felixarntz I was going to mention the spacing between the sentence and link but thought I was being too picky!  😃  It does look a lot better but I agree with @danielgent though it doesn't look aligned.  Shall we open up a new ticket for that, unless we want to change this before the release?

**Comment by felixarntz** [comment]
@wpdarren @danielgent @eugene-manuilov @aaemnnosttv Let's try to fix it before the release on Monday, but without any hacks please :)

We may need to ensure the line height is the same despite the smaller font size? Alternatively, let's change the font size back to what it originally was (i.e. remove the `--tiny` class) - although I like the current version where the link is not that huge.

**Comment by eugene-manuilov** [comment]
@felixarntz do you mind reviewing #3874?

**Comment by felixarntz** [comment]
@eugene-manuilov This now looks good to go on both mobile and desktop after #3874.
